### PR TITLE
Refactoring and splitting basic SymbolLayer icon examples

### PR DESCRIPTION
**Pull request by langsmith**
It's only taken me **two** years to get around to it, but I've always felt like the current `BasicSymbolLayerActivity` example wasn't truly basic. Yes, it added `SymbolLayer` icons, but the example had logic to listen to map clicking, change select state, animate the icon size, etc.

 It's not a barebones and clean example of how to simply add `SymbolLayer` icons to the map with no other frills.

This pr moves the `master` branch's `BasicSymbolLayerActivity` code to an example titled `IconSizeChangeOnClickActivity`  and then cuts out the fluff from the `BasicSymbolLayerActivity`. The `BasicSymbolLayerActivity` now just shows 3 `SymbolLayer` icons in Uruguay.

cc @riastrad and @samfader for mobile support

cc @colleenmcginnis . This refactoring and code movement might make https://docs.mapbox.com/android/maps/examples/ a bit wonky. I will fix things downstream in `/android-docs` once this lands.

ezgif com-resize (6)

### CODE REVIEW

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/BasicSymbolLayerActivity.java [hunk 1]
```diff
@@ -55,117 +54,41 @@ protected void onCreate(Bundle savedInstanceState) {
...
-      if (markerSelected) {
-        deselectMarker(selectedMarkerSymbolLayer);
-      }
-      if (features.size() > 0) {
-        selectMarker(selectedMarkerSymbolLayer);
-      }
-    }
-    return true;
-  }
+    mapboxMap.setStyle(new Style.Builder().fromUrl("mapbox://styles/mapbox/cjf4m44iw0uza2spb3q0a7s41")
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/BasicSymbolLayerActivity.java]: could you add some context why we are using this specific style?

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java [hunk 2]
```diff
@@ -0,0 +1,218 @@
...
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    if (mapboxMap != null) {
+      mapboxMap.removeOnMapClickListener(this);
+    }
+    mapView.onDestroy();
```
Comment (tobrun) [inline MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java]: we aren't stopping animations in this example so we can potentially crash with selecting a symbol and closing the activity quickly after.

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/BasicSymbolLayerActivity.java (hunk 1 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/BasicSymbolLayerActivity.java]: Using a style that isn't found in `Style` class  to just shake things up a little.

Screen Shot 2019-05-13 at 1 02 59 PM

File: MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java (hunk 2 above)
Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java]: Ok, https://github.com/mapbox/mapbox-android-demo/pull/1030/commits/4d4e13a314aca56105a528ec50037e1be0999afb adds animation cancellation in `onDestroy()`.

Comment (langsmith) [reply MapboxAndroidDemo/src/main/java/com/mapbox/mapboxandroiddemo/examples/styles/IconSizeChangeOnClickActivity.java]: I closed the example during the icon size animation. No issues.

**Comment by langsmith** [comment]
Anything else, @tobrun ?

**Pull request merged**